### PR TITLE
Add modular Grok CLI, audio workflows, and proxy updates

### PR DESCRIPTION
**Pull request by 0xSMW**
## Summary

This PR upgrades SwiftGrok from a small client/proxy package into a fuller Grok web client stack: a modular terminal CLI, scriptable automation output, audio transcription workflows, broader Grok web-resource APIs, and a more capable OpenAI-compatible proxy.

## CLI Features

- Replaces the monolithic CLI entry point with focused modules under `Sources/GrokCLI/Commands`, `Core`, `Interactive`, `Parsing`, `Presentation`, and `Runtime`.
- Adds top-level command routing for `chat`, `message`, `transcribe`, `auth`, `list`, `models`/`modes`, `agents`, `tasks`, `skills`, `workspaces`/`workspace`, `files`, and `test`.
- Adds scriptable JSON output across CLI commands through `--json`, `--format json`, and `--format=json`, including consistent result envelopes and NDJSON streaming events for `grok message --stream --json`.
- Adds quiet raw scripting paths so `grok message --raw --quiet` and `grok chat --raw --quiet` keep stdout clean for assistant text while moving UI/status/error noise away from stdout.
- Adds prompt-source support for stdin, explicit `--stdin`, and `--prompt-file`, with validation that inline text, stdin, prompt files, and audio are mutually exclusive.
- Adds audio workflows: `grok transcribe`, `grok message --audio`, `grok chat --audio`, binary audio via `--audio - --audio-format`, interactive `/audio`, `/audio file`, `/audio send`, `/audio-send`, and `/transcribe`.
- Adds local microphone recording for interactive `/audio` on macOS through `ffmpeg` WebM/Opus output, plus `GROK_CLI_AUDIO_DEVICE` for choosing an AVFoundation input.
- Adds Grok web resource commands for agents, tasks, skills, workspaces, and files, including JSON output, help paths, uploads, deletes, conversation-to-workspace linking, and agent instruction editing.
- Improves interactive chat with command parsing, slash/bare command handling, model/workspace/file pickers, attachment management, conversation resume, loaded parent response handling, and cleaner prompt/status behavior.
- Adds model catalog support and aliases for `auto`, `fast`, `expert`, `reasoning`, `think`, `heavy`, `grok-4.3-beta`, `4.3`, `43`, and raw web mode IDs.
- Shows subscription and low rate-limit status in chat so users can see account context and remaining capacity.

## GrokClient Improvements

- Expands `GrokClient` support for current Grok web APIs: modes, subscriptions, rate limits, conversations, responses, tasks, skills, agent settings, workspaces, assets, file upload/delete, and speech-to-text.
- Adds speech-to-text helpers for file paths, raw `Data`, and base64 payloads with format inference for common audio extensions.
- Improves streaming response parsing, final response handling, thinking/search/tool markup parsing, response metadata, web search results, X posts, and fallback behavior.
- Adds `GROK_BASE_URL` and constructor-level base URL normalization so tests and local mock servers can pass a host, `/rest`, or `/rest/app-chat` URL.
- Improves model access errors, unavailable-mode descriptions, and mode availability metadata.
- Redacts sensitive audio/body fields from generated debug/curl output.

## Proxy Features And Fixes

- Adds OpenAI-style model listing on both `/v1/models` and `/models`, backed by the Grok modes catalog instead of a fixed static list.
- Passes resolved Grok modes into chat completions for both streaming and non-streaming proxy requests.
- Adds `/v1/audio/transcriptions` with OpenAI-style multipart upload support, JSON/base64 audio support, text/json response formats, model/language/prompt compatibility fields, and Grok speech-to-text behind it.
- Fixes streaming completion termination by using `ConversationResponse.isFinal`, emitting a terminal chunk with `finish_reason: "stop"`, and still ending with `data: [DONE]`.
- Preserves assistant role emission when thinking chunks arrive before answer chunks.
- Adds request validation for empty messages and invalid chat roles before calling Grok.
- Removes duplicate proxy route registration and improves verbose logging with safer body redaction for audio/content fields.
- Aligns Docker and local proxy credential loading around `GROK_COOKIES` first, then `credentials.json` from the process working directory, then mock fallback cookies.

## Auth, Cookies, Scripts, And Tooling

- Hardens `Scripts/cookie_extractor.py` for browser/profile inference, SQLite/WAL cookie reads, keychain access, required cookie validation, JSON output, and supported browser shortcuts including Safari, Atlas, Chrome, Firefox, Chromium, Brave, Edge, and Arc.
- Adds a legacy root `cookie_extractor.py` wrapper for older workflows.
- Improves `Scripts/install_cli.sh` with `--user`, `--system`, `--bin-dir`, and `--prefix`, and installs the cookie extractor beside the `grok` binary.
- Updates proxy setup/build/run scripts to use the project-root credentials path and current proxy serve command.
- Updates smoke scripts for request, streaming, models, and audio transcription checks.
- Adds VS Code launch/LLDB settings and expands `.gitignore` for local credentials, caches, and generated test artifacts.

## Deprecations And Behavior Changes

- Reasoning is now treated as always enabled for Grok 4 web modes. `--reasoning`, `/reason`, and `/reasoning` remain accepted as legacy no-op compatibility controls and warn instead of changing payloads.
- `--deep-search` and `--no-search` remain accepted as no-op compatibility shims; search is automatic and no longer sent as a configurable Grok payload field.
- `--no-custom-instructions`, local custom-instruction overrides, and Grok 3 personality/system prompt presets are retained only for compatibility. Instructions now live in Grok agent settings and deprecated custom instruction fields are no longer sent to Grok.
- OpenAI-style proxy fields such as `temperature`, system messages, and transcription model/language/prompt are accepted for client compatibility but are not mapped one-for-one to Grok controls when Grok does not expose equivalent behavior.
- Bare `model 4` is not documented as an alias. Users should use `model 4.3`, `model grok-4.3-beta`, `43`, or the raw web mode ID `grok-420-computer-use-sa`.
- JSON mode reserves stdout for machine-readable JSON. Human banners, ANSI output, progress, and debug text are suppressed or moved out of stdout.

## Documentation

- Rewrites the main README around install, authentication, advanced configuration, JSON/quiet scripting, audio/transcription, models, agents, workspaces, tasks, skills, files, conversations, package usage, proxy usage, and troubleshooting.
- Updates `PROXY_README.md` for current setup, endpoints, streaming, audio transcription, model listing, credentials, environment variables, and OpenAI-compatibility notes.
- Updates `DOCKER.md` to match actual Docker credential precedence and explain host-side credential generation and audio transcription testing.
- Adds `Tests/README.md` with Swift test commands, Python cookie extractor tests, and proxy smoke commands.
- Adds `RELEASES.md` notes for `0.2.0` and Grok 4 compatibility changes.
- Adds planning specs under `plans/` for CLI JSON mode, module decomposition, command behavior, scripting gaps, interactive input fixes, audio input, terminal UI quality, and proxy/docs streaming fixes.

## Test Coverage Added

- Adds `GrokCLIE2ETests` for top-level routing, help/errors, JSON and NDJSON output, quiet/raw output boundaries, prompt files, stdin, auth generation/import, audio/transcription, interactive slash commands, model selection, resource commands, conversation resume, formatting, and rate-limit display.
- Expands `GrokClientTests` for speech-to-text, endpoint normalization, request payloads, model aliases, mode/subscription/rate-limit decoding, stream parsing, final response behavior, and sensitive body redaction.
- Expands `GrokProxyTests` for `/v1/models`, `/models`, chat validation, streaming terminal chunks, audio transcription validation, response formats, and multipart conversion.
- Adds `Tests/test_cookie_extractor.py` for browser cookie extraction edge cases and credential output behavior.
- Updates `Scripts/test_proxy_streaming.sh` to assert both terminal `finish_reason: "stop"` chunks and final `data: [DONE]` markers.

## Validation

Not run while preparing this PR update, per request.

### CODE REVIEW

File: Sources/GrokCLI/Parsing/ModelOptionParsing.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+}
+
+extension GrokCLI {
+    static func applyModelOption(_ arg: String, nextValue: String?) -> (mode: GrokMode?, consumedNext: Bool, missingValue: Bool) {
+        if arg == "--model" || arg == "--mode" {
+            guard let nextValue, !nextValue.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
+                return (nil, false, true)
+            }
+            return (GrokMode.resolve(nextValue), true, false)
```
[the remark this answers is not in the document]
Comment (0xSMW) [reply Sources/GrokCLI/Parsing/ModelOptionParsing.swift]: Fixed in `30490e3`: `--model` / `--mode` now treat a following option-looking token as a missing model value instead of consuming it as a custom mode. Added a CLI E2E regression for `grok message --model --json hello` to verify it exits with a usage error and does not send a request.

File: Sources/GrokCLI/Core/TopLevelRouter.swift [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+            "--no-search",
+            "--markdown",
+            "-m",
+            "--raw",
+            "--json",
+            "--debug",
+            "--no-custom-instructions",
+            "--private",
+            "--stream"
+        ].contains(arg)
```
[the remark this answers is not in the document]
Comment (0xSMW) [reply Sources/GrokCLI/Core/TopLevelRouter.swift:170]: Fixed in `30490e3`: leading `--quiet` is now included in top-level option normalization, and `--prompt-file` is normalized as a leading value option too. Added a CLI E2E regression for `grok --quiet --raw message hello` to verify it dispatches to the one-shot message command and keeps stdout answer-only.

File: Tests/GrokCLIE2ETests/GrokCLIE2ETests.swift [hunk 3]
```diff
@@ -0,0 +1,2550 @@
+import Foundation
+import Network
```
[the remark this answers is not in the document]
Comment (0xSMW) [reply Tests/GrokCLIE2ETests/GrokCLIE2ETests.swift]: Fixed in `30490e3`: the Network-backed E2E implementation is now guarded with `#if canImport(Network)`, with a skip-only fallback test on platforms without Network. This keeps the macOS E2E suite intact while avoiding a Linux compile failure from the unconditional import.

File: Sources/GrokProxy/Controllers/AudioTranscriptionsController.swift [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        )
+    }
+
+    private func normalizeResponseFormat(_ responseFormat: String?) -> String {
+        responseFormat?.lowercased() ?? "json"
+    }
+
+    private func inferAudioFormat(from file: File) -> String? {
+        if let extensionName = file.filename.split(separator: ".").last {
+            return String(extensionName).lowercased()
```
[the remark this answers is not in the document]
Comment (0xSMW) [reply Sources/GrokProxy/Controllers/AudioTranscriptionsController.swift]: Fixed in `30490e3`: multipart transcription now only uses a filename extension when one exists, and otherwise parses the multipart part headers to fall back to the file part `Content-Type` subtype. Added a proxy regression for `filename="recording"` with `Content-Type: audio/wav` to verify `audioFormat` becomes `wav`.
